cinema/views.py

```diff
@@ -118,12 +156,25 @@
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get_permissions(self):
+        if self.action == "list" and self.request.user.is_authenticated:
+            self.permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+        if self.action == "delete":
+            self.permission_classes = ()
```

Comment:
Similar to `GenreViewSet`, setting `permission_classes` to an empty tuple in the `delete` action of `OrderViewSet` removes all permissions. Ensure this is the desired behavior or specify the correct permissions.